### PR TITLE
[446] Add progress/state tracking for programme-details section

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/x9qJPXb5/446-programme-details-progress-state

### Changes proposed in this pull request

- Add the progress/state for the programme details section

### Guidance Review

- Visit the review app - https://dfe-twd-rttd-pr-176.herokuapp.com/
- Assert summary page shows 'Not started'
- Fill out some details but do not mark complete, assert label shows 'in progress'
- Mark complete, assert label shows 'completed'

### CODE REVIEW

**Comment by darokel** [comment]
That watermark is very sneaky..

**Comment by darokel** [comment]
[Programme details progress/state](https://trello.com/c/x9qJPXb5/446-programme-details-progress-state)

File: spec/features/trainees/edit_programme_details_spec.rb [hunk 1]
```diff
@@ -43,6 +81,16 @@ def and_i_submit_the_form
     @programme_details_page.submit_button.click
   end
 
+  def and_i_visit_the_summary_page
+    @summary_page ||= PageObjects::Trainees::Summary.new
```
Comment (jhenderson) [inline spec/features/trainees/edit_programme_details_spec.rb:85]: You can get rid of all instances of `@summary_page ||= PageObjects::Trainees::Summary.new` and just use the attr reader `summary_page` which is declared in `spec/support/features/trainee_steps.rb`

File: spec/support/features/trainee_steps.rb [hunk 2]
```diff
@@ -29,5 +29,13 @@ def summary_page
     def confirm_details_page
       @confirm_page ||= PageObjects::Trainees::ConfirmDetails.new
     end
+
+    def and_i_confirm_my_details(checked: true, section:)
+      checked_option = checked ? "check" : "uncheck"
+      confirm_details_page ||= PageObjects::Trainees::ConfirmDetails.new
```
Comment (jhenderson) [inline spec/support/features/trainee_steps.rb:35]: `confirm_details_page` is already defined above. If you remove it the tests still pass.

**Review by jhenderson — CHANGES_REQUESTED** [review]
Requested some minor tweaks, otherwise looks good.

File: spec/support/features/trainee_steps.rb (hunk 2 above)
Comment (darokel) [reply spec/support/features/trainee_steps.rb]: good spot https://github.com/DFE-Digital/register-trainee-teachers/pull/176/commits/7e11f2317b5e083113e8eca1b35f16704c43452b

File: spec/features/trainees/edit_programme_details_spec.rb (hunk 1 above)
Comment (darokel) [reply spec/features/trainees/edit_programme_details_spec.rb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/176/commits/7e11f2317b5e083113e8eca1b35f16704c43452b

**Review by jhenderson — APPROVED** [review]

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**